src/lib.rs

```diff
@@ -262,20 +262,125 @@ impl<T> StableVec<T> {
     /// This method uses exactly the same deletion strategy as
     /// [`remove()`](#method.remove).
     ///
+    /// # Example
+    ///
+    /// ```
+    /// # use stable_vec::StableVec;
+    /// let mut sv = StableVec::from(&[1, 2, 3]);
+    /// assert_eq!(sv.pop(), Some(3));
+    /// assert_eq!(sv.into_vec(), vec![1, 2]);
+    /// ```
+    ///
     /// # Note
     ///
     /// This method needs to find index of the last valid element. Finding it
     /// has a worst case time complexity of O(n). If you already know the
     /// index, use [`remove()`](#method.remove) instead.
     pub fn pop(&mut self) -> Option<T> {
-        let last_index = self.deleted
-            .iter()
-            .enumerate()
-            .rev()
-            .find(|&(_, deleted)| !deleted)
-            .map(|(i, _)| i)
-            .unwrap_or(0);
-        self.remove(last_index)
+        self.find_last_index().and_then(|index| self.remove(index))
+    }
+
+    /// Removes and returns the first element from this collection, or `None` if
+    /// it's empty.
+    ///
+    /// This method uses exactly the same deletion strategy as
+    /// [`remove()`](#method.remove).
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use stable_vec::StableVec;
+    /// let mut sv = StableVec::from(&[1, 2, 3]);
+    /// assert_eq!(sv.pop_first(), Some(1));
+    /// assert_eq!(sv.into_vec(), vec![2, 3]);
+    /// ```
+    ///
+    /// # Note
+    ///
+    /// This method needs to find index of the first valid element. Finding it
+    /// has a worst case time complexity of O(n). If you already know the
+    /// index, use [`remove()`](#method.remove) instead.
+    pub fn pop_first(&mut self) -> Option<T> {
+        self.find_first_index().and_then(|index| self.remove(index))
+    }
+
+    /// Finds the first element and returns a reference to it, or `None` if it's empty.
+    ///
+    /// This method has O(n) complexity.
```

Comment:
I think it's worth mentioning that this is a worst case. Usually, stable vectors are pretty densely packed (otherwise a lot of memory is used), so most of the time the element is found after a few slots. Just saying "O(n) complexity" sounds like it always does n steps. Why not also copy the note from the functions above here?

Reply:
[Big O notation](https://en.wikipedia.org/wiki/Big_O_notation) describes the growth rate of a function and implies the upper bound, therefore:
* Assuming anything other than that is incorrect
* Prefixing this notation with "worst case" is pretty much a tautology
That's how I understand it, anyway.

Reply:
Absolutely: the statement is correct as it is. As is the statement "This method has O(n³) complexity". And yes, adding "worst case" is technically a tautology. 

*However*, very often, O(f) is used and understood as Θ(f) (meaning: exactly always this runtime complexity). This is not correct, but it's how many people think about it. That's the reason I prefer the more verbose explanation of the function's complexity. I'd rather explain a bit more in the documentation for humans than to rely on "technically correct".

Reply:
I'm still not convinced that we should cater to unprofessional programmers, but I will fix comments anyway. After all, it's your library, and I can always maintain my own fork, if I want to be that pedantic (no way, ha-ha).

Reply:
Thanks :)